echoregions/utils/api.py

```diff
@@ -2,40 +2,58 @@
 
 import numpy as np
 import xarray as xr
-from xarray import DataArray
+from xarray import DataArray, Dataset
 
 from ..regions2d.regions2d import Regions2D
 
 
-def convert_mask_2d_to_3d(M: DataArray):
-    """Convert 2D Mask data into its 3D one-hot encoded form.
+def convert_mask_2d_to_3d(mask_2d_da: DataArray) -> Dataset:
+    """Convert 2D multi-labeled mask data into its 3D one-hot encoded form.
     Parameters
     ----------
-    M : Data Array
-        A DataArray with the data_var masked by a specified region.
+    mask_2d_da: DataArray
+        A DataArray with the data_var masked by a specified region. This data will
+        be in the form of integer, demarking labels of masked regions, and nan values,
+        demarking non-masked areas.
     Returns
     -------
-    A Dataset with a 3D DataArray with the data_var masked by the specified
-    region in one-hot encoded form and a dictionary to remember original
-    non-nan values.
+    mask_3d_ds : Dataset
+        A Dataset with a 3D DataArray/mask and each layer of the 3D mask will contain
+        a 1s/0s mask for each unique label in the 2D mask. The layers will be labeled
+        by a dictionary that maps the individual label layers of the 3D mask to an integer
+        label in the 2D mask.
+    Notes
+    -----
+    -1 in the dictionary of mask_3d_ds means that there exists no masked
```

Comment:
I think empty data array would be better: `xr.DataArray(data=None)` and `.isnull()`